lib/embulk/output/bigquery/gcs_client.rb

```diff
@@ -25,7 +25,19 @@ def insert_bucket(bucket = nil)
           begin
             Embulk.logger.info { "embulk-output-bigquery: Insert bucket... #{@project}:#{bucket}" }
             body = {
-              name: bucket
+              name: bucket,
+              lifecycle: {
+                rule: [
+                  {
+                    action: {
+                      type: "Delete",
+                    },
+                    condition: {
+                      age: 1,
+                    }
+                  },
+                ]
+              }
```

Comment:
1. Renaming the method to “insert_temporary_bucket”, or
2. Accept any lifecycle option and pass the value from caller

is preferable. I forgot which way I assumed when I created this for extending.

Reply:
1 should be fine because this Gem doesn’t aim to create generic gcs client.